pkg/build/api/v1/types.go

```diff
@@ -658,6 +734,27 @@ type BuildRequest struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar `json:"env,omitempty"`
+
+	// BuildCauses can pass information from the controllers to the buildconfig
+	// generator about who/what has triggered a build through the request.
+	BuildCauses []*BuildTriggerCauses `json:"BuildCauses,omitempty"`
+}
+
+// BuildTriggerCauses passes information from the controllers to the
+// buildconfig generator about who/what has triggered a build
+type BuildTriggerCauses struct {
```

Comment:
sorry, this isn't redundant, I missed that i transitioned into the v1 types file, nm.